File: 0013-roman-to-integer/0013-roman-to-integer.cs
public class Solution {
    public int RomanToInt(string s) {
        s = s.Replace("CM","DCCCC").Replace("CD","CCCC").Replace("XL","XXXX").Replace("XC","LXXXX").Replace("IV","IIII").Replace("IX","VIIII");
        int sum = 0;
        foreach (char i in s){
            if(i == 'M'){
                sum += 1000;
            }
            else if(i == 'D'){
                sum += 500;
            }
            else if(i == 'C'){
                sum += 100;
            }
            else if(i == 'L'){
                sum += 50;
            }
            else if(i == 'X'){
                sum += 10;
            }
            else if(i == 'V'){
                sum += 5;
            }
            else if(i == 'I'){
                sum += 1;
            }
            else{
                continue;
            }
        }

        return sum;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        value_symbols = {
            'M':1000,
            'D':500,
            'C':100,
            'L':50,
            'X':10,
            'V':5,
            'I':1
        }

        ans = 0

        for i in range(1, len(s)+1):
            if i < len(s) and value_symbols[s[i-1]] < value_symbols[s[i]]:
                ans -= value_symbols[s[i-1]]
            else:
                ans += value_symbols[s[i-1]]
        
        return ans
